1188: Adds ability to jump to a new neighborhood

nomichka opened the pull request:
Resolves #1188 

Pressing the "Jump" icon goes to this screen:
![Project Sidewalk Issue 1188 (2)](https://user-images.githubusercontent.com/35845622/61011670-6cc33180-a330-11e9-9588-04d27d63aadd.PNG)

Clicking on "Go to a new location in this neighborhood" goes to:
![Project Sidewalk Issue 1188 (3)](https://user-images.githubusercontent.com/35845622/61011850-0e4a8300-a331-11e9-9f5d-73f418b5c02a.PNG)

Clicking on "Go to a new neighborhood" has the same effect as selecting the "Go to a new neighborhood" option when finishing a neighborhood.

misaugstad commented:
@nomichka this works well! can you just make the decision of new neighborhood vs. current neighborhood options look a bit more like buttons?

nomichka commented:
@misaugstad Sure! The options look like buttons now.

When not hovering over a button:
![Project Sidewalk Issue 1188 unhover](https://user-images.githubusercontent.com/35845622/61671676-84d67180-ac9d-11e9-9eab-3b0c4948d377.PNG)

When hovering over a button:
![Project Sidewalk Issue 1188 hover](https://user-images.githubusercontent.com/35845622/61671681-89028f00-ac9d-11e9-9530-1ac20f2c64b3.PNG)

misaugstad commented:
Ok I've been thinking about this a bit more carefully...

1. I think it makes more sense to have the "explore a new area" vs "can't go walk this direction" question come first, then if they say they want to explore a new area, you ask if they want to stay in the same neighborhood or go to a new neighborhood
1. The two popups should have the same look and feel. I like your use of buttons better than the current radio button + OK button. Could you change that one to look more like yours?
1. We should be logging whenever someone clicks a button or anything. Could you add some logging for your new buttons?

nomichka commented:
I made the requested changes, and the screen first looks like this when you press the jump button:
![Jump to another](https://user-images.githubusercontent.com/35845622/62882039-e65e8e80-bce5-11e9-82c2-d84188b4277f.PNG)
Where if you press "I cannot go..." you're transported to another location, and if you press "I want to..." you get this screen:
![Go to a new](https://user-images.githubusercontent.com/35845622/62882161-23c31c00-bce6-11e9-8113-9bb55aab0aa3.PNG)
Where the buttons are self-explanatory.

Also, because there's no cancel button anymore, I changed it so pressing on the modal background at either screen has the same effect as the cancel button.

I'm pretty sure I changed it so it's logging, but I've never worked with the logging, so I'm not completely certain on that front.

jonfroehlich commented:
I think removing the Cancel button may be problematic (did the original UIs
have those?) because implicit cancels actions can be confusing for users.

On Mon, Aug 12, 2019 at 9:53 AM Naomi Bashkansky <notifications@github.com>
wrote:

> I made the requested changes, and the screen first looks like this when
> you press the jump button:
> [image: Jump to another]
> <https://user-images.githubusercontent.com/35845622/62882039-e65e8e80-bce5-11e9-82c2-d84188b4277f.PNG>
> Where if you press "I cannot go..." you're transported to another
> location, and if you press "I want to..." you get this screen:
> [image: Go to a new]
> <https://user-images.githubusercontent.com/35845622/62882161-23c31c00-bce6-11e9-8113-9bb55aab0aa3.PNG>
> Where the buttons are self-explanatory.
>
> Also, because there's no cancel button anymore, I changed it so pressing
> on the modal background at either screen has the same effect as the cancel
> button.
>
> I'm pretty sure I changed it so it's logging, but I've never worked with
> the logging, so I'm not completely certain on that front.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ProjectSidewalk/SidewalkWebpage/pull/1786?email_source=notifications&email_token=AAML55NDW25SO7KHYHZDBRLQEGIPJA5CNFSM4IAC4MEKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4DES2I#issuecomment-520505705>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAML55MKAYIXUQ7QN4CWRDLQEGIPJANCNFSM4IAC4MEA>
> .
>


-- 
Jon Froehlich
Associate Professor
Paul G. Allen School of Computer Science & Engineering
University of Washington
http://makeabilitylab.io
@jonfroehlich <https://twitter.com/jonfroehlich> - Twitter
Help make sidewalks more accessible: http://projectsidewalk.io

misaugstad commented:
Yeah I agree that a cancel button on both of those are probably a good idea

nomichka commented:
I'm thinking to have the cancel button on the bottom-right to space things out a bit:
![Cancel button](https://user-images.githubusercontent.com/35845622/62983156-3ec78600-bde3-11e9-968b-ecbb28ac0d2d.PNG)
![Cancel2button](https://user-images.githubusercontent.com/35845622/62983254-8d752000-bde3-11e9-93a3-60686d6025be.PNG)
Is that good?

For reference, the feedback screen looks like this:
![Feedback](https://user-images.githubusercontent.com/35845622/62983314-c7462680-bde3-11e9-8d58-e865adcdedb8.PNG)

misaugstad commented:
Looks good to me!
